src/alchemlyb/parsing/lammps.py

```diff
@@ -0,0 +1,1303 @@
+            )
+        )
+
+    return lambda_values, lambda_pairs, lambda2
+
+
+@_init_attrs
+def extract_u_nk_from_u_n(
+    fep_files,
+    T,
+    column_lambda,
+    column_U,
+    column_U_cross,
+    dependence=lambda x: (x),
+    index=-1,
+    units="real",
+    prec=4,
+    ensemble="nvt",
+    pressure=None,
+    column_volume=4,
+):
+    """Produce u_nk from files containing u_n given a separable dependence on lambda.
+
+    Parameters
+    ----------
+    fep_files : str or list
+        If not a list, a str representing the path to fepout file(s) to extract data from. Filenames and paths are
+        aggregated using `glob <https://docs.python.org/3/library/glob.html>`_. For example, "/path/to/files/something_*.txt".
+    T : float
+        Temperature in Kelvin at which the simulation was sampled.
+    columns_lambda : int
+        Indices for columns (file column number minus one) representing the lambda at which the system is equilibrated
+    column_U : int
+        Index for the column (file column number minus one) representing the potential energy of the system.
+    column_U_cross : int
+        Index for the column (file column number minus one) representing the potential energy of the cross interactions
+        between the solute and solvent.
+    dependence : func, default=`lambda x : (x)`
+        Dependence of changing variable on the potential energy, which must be separable.
+    index : int, default=-1
+        In provided file names, using underscore as a separator, these indices mark the part of the filename
+        containing the lambda information for :func:`alchemlyb.parsing._lambda_from_filename`. If ``column_lambda2 != None``
+        this list should be of length three, where the last value represents the invariant lambda.
+    units : str, default="real"
+        Unit system used in LAMMPS calculation. Currently supported: "cgs", "electron", "lj". "metal", "micro", "nano",
+        "real", "si"
+    prec : int, default=4
+        Number of decimal places defined used in ``round()`` function.
+    ensemble : str, default="nvt"
+        Ensemble from which the given data was generated. Either "nvt" or "npt" is supported where values from NVT are
+        unaltered, while those from NPT are corrected
+    pressure : float, default=None
+        The pressure of the system in the NPT ensemble in units of energy / volume, where the units of energy and volume
+        are as recorded in the LAMMPS dump file.
+    column_volume : int, default=4
+        The column for the volume in a LAMMPS dump file.
+
+    Returns
+    -------
+    u_nk_df : pandas.Dataframe
+        Dataframe of potential energy for each alchemical state (k) for each frame (n).
+        Note that the units for timestamps are not considered in the calculation.
+
+        Attributes
+
+        - temperature in K
+        - energy unit in kT
+
+    .. versionadded:: 2.4.1
+
+    """
+    # Collect Files
+    if isinstance(fep_files, list):
```

Comment:
Does this function needs to see all the files or it could be separated? For example, could you do
```
u_nk_list = [extract_u_nk_from_u_n(filepath) for filepath in filepath_list]
u_nk = pd.concat(u_nk_list)
```
Instead of `u_nk = extract_u_nk_from_u_n(filepath_list)`
The reason that I'm asking this as if you could load the energy file by file and then concatenate it, this step could be parallelised, otherwise, it has to be loaded file by file.